src/Controls/Internal/TextCopyPasteHelper.ts

```diff
@@ -0,0 +1,87 @@
+﻿module Fayde.Controls.Internal {
+    export class TextCopyPasteHelper {
+        private callback: any = false;
+        private pastedText: string = "";
+
+		GetPastedText(): string {
+            return this.pastedText;
+        }
+
+        private createElement(text: string): HTMLDivElement {
+            (<any>document.body).scroll = "no";
+            document.documentElement.style.overflow = "hidden";
+            var div = document.createElement("div");
+            div.id = "special_copy";
+            div.style.left = "-1000px";
+            div.style.top = "-1000px";
+            div.style.position = "absolute";
+            div.textContent = text;
+            document.body.appendChild(div);
+            div.contentEditable = "true";
+            return div;
+        }
+
+        private selectContent(element: HTMLDivElement) {
+            var rangeToSelect = document.createRange();
+            rangeToSelect.selectNodeContents(element);
+
+            // select the contents
+            var selection = window.getSelection();
+            selection.removeAllRanges();
+            selection.addRange(rangeToSelect);
+        }
+
+        CopyText(text: string): void {
+            var win = <any>window;
+
+            if (win.clipboardData) { // internet explorer
+                var res = win.clipboardData.setData("Text", text);
+                if (!res)
+                    alert("Your browser do not allow copy to the clipboard");
+            } else {
+                var div = this.createElement(text);
+                this.selectContent(div);
+
+                if (win.netscape && netscape.security) {
+                    netscape.security.PrivilegeManager.enablePrivilege("UniversalXPConnect");
+                }
+
+                // Copy the selected content to the clipboard
+                // Works in Firefox and in Safari before version 5
+                if (!document.execCommand("copy", false, null))
+                    alert("Your browser does not allow copy to the clipboard. This feature will not function");
+
+                document.body.removeChild(div);
```

Comment:
We are already taking advantage of a helper being isolated away from a textbox.  First, we could build browser-specific interops that follow a very simple interface.
```
interface IClipboardHelper {
  Copy(text: string);
  GetContents(callback: (text: string) => void);
}
```

Next, you can use [`nullstone.Memoizer`](https://github.com/wsick/nullstone/blob/master/src/Memoizer.ts) providing a creator function.  The `memoize` function, when called, will call the creator if not established.  Otherwise, it will return the existing object.